presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkOutputBuffer.java

```diff
@@ -33,8 +35,12 @@
     @GuardedBy("monitor")
     private boolean finished;
 
-    public PrestoSparkOutputBuffer(OutputBufferMemoryManager memoryManager)
+    private final AtomicLong totalRowsProcessed = new AtomicLong();
+    private final AtomicLong totalPagesProcessed = new AtomicLong();
+
+    public PrestoSparkOutputBuffer(String type, OutputBufferMemoryManager memoryManager)
```

Comment:
nit: looks like only "ROW" and "PAGE" can be valid input to this String, what about make it to be an enum?

I am asking this because I originally thought it's a Presto type (e..g  `BIGINT`, `ARRAY(VARCHAR)` )

Reply:
Technically it can accept anything that implements `PrestoSparkBufferedResult`. What do you think about renaming `type` to something like `description`?

Reply:
Actually I removed this fields from the `PrestoSparkOutputBuffer` class and added private `enum OutputBufferType` in the `PrestoSparkTaskExecutorFactory`